src/Mod/Sketcher/Gui/CommandSketcherTools.cpp

```diff
@@ -983,50 +983,61 @@ void CmdSketcherRestoreInternalAlignmentGeometry::activated(int iMsg)
 
     getSelection().clearSelection();
 
-    // go through the selected subelements
-    for (std::vector<std::string>::const_iterator it=SubNames.begin(); it != SubNames.end(); ++it) {
-        // only handle edges
-        if ((it->size() > 4 && it->substr(0,4) == "Edge") ||
-            (it->size() > 12 && it->substr(0,12) == "ExternalEdge")) {
-            int GeoId;
-            if (it->substr(0,4) == "Edge")
-               GeoId = std::atoi(it->substr(4,4000).c_str()) - 1;
-            else
-               GeoId = -std::atoi(it->substr(12,4000).c_str()) - 2;
-
-            const Part::Geometry *geo = Obj->getGeometry(GeoId);
-            // Only for supported types
-            if (geo->getTypeId() == Part::GeomEllipse::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfEllipse::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfHyperbola::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfParabola::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomBSplineCurve::getClassTypeId()) {
+    std::vector<int> SubGeoIds(SubNames.size());
+    std::transform(SubNames.begin(), SubNames.end(), SubGeoIds.begin(),
+                   [Obj](const std::string& SubName) {
+                       int GeoId;
+                       Sketcher::PointPos PosId;
+                       getIdsFromName(SubName, Obj, GeoId, PosId);
+                       return GeoId;
+                   });
```

Comment:
Could you define the lambda out of the call (and add a small comment about what it does) to improve readability ?

Also I wonder if capturing `Obj` by reference won't be stack friendly (and overall improve performance) ???

Reply:
Will update the lambda in the next force push. `Obj` itself is a pointer so I don't understand expect much change, but capture by reference makes more sense, I suppose.

Reply:
No, you're right. As it's a pointer, AFAIK it will be captured as a reference to the value. ;)

Reply:
Changes made. I still went with `[&Obj]` even if it makes no difference.